src/Plugin/RemoteCheck/RemoteRegister.php

```diff
@@ -0,0 +1,207 @@
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\relaxed\Entity\RemoteInterface;
+use Drupal\relaxed\Plugin\RemoteCheckBase;
+use Drupal\relaxed\SensitiveDataTransformer;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Psr7\Uri;
+use GuzzleHttp\RequestOptions;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * @RemoteCheck(
+ *   id = "remote_register",
+ *   label = "Register on remote",
+ * )
+ */
+class RemoteRegister extends RemoteCheckBase implements ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The config factory service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * The serializer.
+   *
+   * @var \Symfony\Component\Serializer\Serializer
+   */
+  protected $serializer;
+
+  /**
+   * The Guzzle HTTP client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  protected $httpClient;
+
+  /**
+   * The related request stack.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * The sensitive data transformer.
+   *
+   * @var \Drupal\relaxed\SensitiveDataTransformer
+   */
+  protected $sensitiveDataTransformer;
+
+  /**
+   * The Messenger service.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * The Entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * RemoteRegister constructor.
+   *
+   * @param array $configuration
+   * @param $plugin_id
+   * @param $plugin_definition
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   * @param \Symfony\Component\Serializer\Serializer $serializer
+   * @param \GuzzleHttp\ClientInterface $http_client
+   * @param \Symfony\Component\HttpFoundation\RequestStack $request_stack
+   * @param \Drupal\relaxed\SensitiveDataTransformer $sensitive_data_transformer
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory, Serializer $serializer, ClientInterface $http_client, RequestStack $request_stack, SensitiveDataTransformer $sensitive_data_transformer, MessengerInterface $messenger, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $config_factory;
+    $this->serializer = $serializer;
+    $this->httpClient = $http_client;
+    $this->requestStack = $request_stack;
+    $this->sensitiveDataTransformer = $sensitive_data_transformer;
+    $this->messenger = $messenger;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * @param \Drupal\contentpool_client\Plugin\RemoteCheck\ContainerInterface $container
```

Comment:
Misses param descriptions....  Misses proper comment. misses return comment. -> this must pass CS checks!